File: app/src/main/java/com/karanchuk/roman/testtranslate/data/database/model/Language.java
package com.karanchuk.roman.testtranslate.data.database.model;

import android.support.annotation.NonNull;

/**
 * Created by roman on 18.4.17.
 */

public class Language implements Comparable<Language>{
    private String mName;
    private String mAbbr;
    private boolean mSelected;

    public Language(final String name,
                    final String abbr,
                    final boolean selected) {
        mName = name;
        mAbbr = abbr;
        mSelected = selected;
    }

    public String getName() {
        return mName;
    }

    public void setName(final String name) {
        mName = name;
    }

    public boolean isSelected() {
        return mSelected;
    }

    public void setSelected(final boolean selected) {
        mSelected = selected;
    }

    @Override
    public int compareTo(@NonNull final Language o) {
        return mName.compareTo(o.mName);
    }

    @Override
    public String toString() {
        return mName;
    }

    public String getAbbr() {
        return mAbbr;
    }

    public void setAbbr(final String abbr) {
        mAbbr = abbr;
    }

    @Override
    public boolean equals(final Object obj) {
        if (obj instanceof Language) {
            Language item = (Language) obj;
            if (mAbbr.equals(item.mAbbr)) {
                return true;
            }
        }
        return false;
    }
}

File: app/src/main/java/com/romankaranchuk/translator/data/database/model/Language.kt
package com.romankaranchuk.translator.data.database.model

import java.util.Objects

/**
 * Created by roman on 18.4.17.
 */
data class Language(
    var name: String,
    var abbr: String,
    var isSelected: Boolean
) : Comparable<Language> {

    override fun compareTo(o: Language): Int {
        return name.compareTo(o.name)
    }

    override fun toString(): String {
        return name
    }

    override fun hashCode(): Int {
        return Objects.hash(abbr)
    }

    override fun equals(obj: Any?): Boolean {
        if (obj is Language) {
            return Objects.equals(abbr, obj.abbr)
        }
        return false
    }
}